[REF] ActionMapping - Improve class structure

colemanw opened the pull request:
Overview
------------------

Completes the refactoring that began with https://github.com/civicrm/civicrm-core/pull/6687 (yes that was in 2015... better late than never).

Technical Details
---------

- Migrates from the old model based on the ([recently dropped](https://github.com/civicrm/civicrm-core/pull/26646)) `civicrm_action_mapping` table by removing all references to that structure and fully embracing the object-model.
- New common base-class added to make it easer to implement `MappingInterface`.
- All functionality should remain unchanged.

colemanw committed:
ActionMapping - Rename ambiguous 'getEntity' function

Before: function getEntity() actually returns the name of the table, not the entity
After: function getEntityTable() returns table name and getEntityName() returns entity name

totten commented on Civi/ActionSchedule/MappingInterface.php:
```diff
@@ -23,9 +23,16 @@ interface MappingInterface {
+   * Main entity name e.g. `Activity`
+   * @return string
+   */
+  public function getEntityName(): string;
```
Regarding `universe` - I don't see any published/contrib things that reference `MappingInterface`. So splitting this method (`getEntity()` => `getEntityTable()`+`getEntityName()`) should be OK.

colemanw replied:
In my next commit i'm adding the `getEntity` function back in as a noisily-deprecated alias of `getEntityTable`. Just in case.

colemanw committed:
ActionMapping - Improve class structure

Completes the refactoring that began with https://github.com/civicrm/civicrm-core/pull/6687
Fully migrates from the old model based on the civicrm_action_mapping table
(dropped in https://github.com/civicrm/civicrm-core/pull/26646)
by removing all references to that structure and fully embracing the object-model.
New common base-class added to make it eaiser to implement MappingInterface.

colemanw committed:
ActionMapping - Clean up obtuse methods of fetching a MappingInterface class

Changed calls to getMappings to use getMapping if they only needed a single value.
In some cases they could be entirely removed because they weren't using the retrieved class for anything.
Deprecated the use of 'filters' param in getMappings as it was redundant with simply calling getMapping with an id.

colemanw commented:
Unrelated failure. Retest this please.

seamuslee001 commented:
This seems fine to me and the test coverage here should be sufficient to guard against issues
